lib/buffer.js

```diff
@@ -276,23 +276,27 @@ function byteLength(string, encoding) {
   if (typeof(string) !== 'string')
     string = String(string);
 
-  switch (encoding) {
-    case 'ascii':
-    case 'binary':
-    case 'raw':
-      return string.length;
+  if (string.length > 0) {
```

Comment:
Maybe just do `if (string.length === 0) return 0;` here? That would trim down the diff a bunch. :)